src/main/java/sirius/biz/tenants/UserAccountData.java

```diff
@@ -110,28 +109,35 @@ public UserAccountData(BaseEntity<?> userObject) {
 
     @BeforeSave
     protected void verifyData() {
-        if (Strings.isFilled(email)) {
-            if (ms.isValidMailAddress(email.trim(), null)) {
-                email = email.toLowerCase();
-            } else {
-                throw Exceptions.createHandled().withNLSKey("Model.invalidEmail").set("value", email).handle();
-            }
-        }
+        userObject.ifChangedAndFilled(UserAccount.USER_ACCOUNT_DATA.inner(UserAccountData.PERSON)
+                                                                   .inner(PersonData.SALUTATION),
+                                      getPerson()::verifySalutation);
+
+        userObject.ifChangedAndFilled(UserAccount.USER_ACCOUNT_DATA.inner(UserAccountData.EMAIL), () -> {
+            email = email.trim().toLowerCase();
+            ms.failForInvalidEmail(email, null);
+        });
+
+        fillAndVerifyUsername();
+    }
 
+    private void fillAndVerifyUsername() {
+        // Use email address if no explicit username is present
         if (Strings.isEmpty(getLogin().getUsername())) {
             getLogin().setUsername(getEmail());
         }
-        if (Strings.isFilled(getLogin().getUsername())) {
-            getLogin().setUsername(getLogin().getUsername().toLowerCase());
-        } else {
-            throw Exceptions.createHandled()
-                            .withNLSKey("Property.fieldNotNullable")
-                            .set("field", NLS.get("LoginData.username"))
-                            .handle();
-        }
 
-        userObject.assertUnique(UserAccount.USER_ACCOUNT_DATA.inner(LOGIN).inner(LoginData.USERNAME),
-                                getLogin().getUsername());
+        // Ensure that the username is is filled and unique...
+        userObject.verifyIfChangedFailIfEmpty(UserAccount.USER_ACCOUNT_DATA.inner(LOGIN).inner(LoginData.USERNAME),
+                                              () -> {
+                                                  // Make it lowercase...
+                                                  getLogin().setUsername(getLogin().getUsername().toLowerCase());
```

Comment:
Requires a check if already saved data is lowercase (and a batch-convert) when not?

Reply:
this code isn't new, it was just moved to another place...